### PR TITLE
fix(codeowners): Start task after a delay and retry once on error

### PR DESCRIPTION
**Pull request by NisanthanNanthakumar**
## Objective:
Fixes [SENTRY-TH3](https://sentry.io/organizations/sentry/issues/3084878291/?project=1&referrer=jira_integration)

It looks like the task is failing to find the Commit before it has been saved. Since we do not make any realtime guarantees for this feature, I added a delay to when we start the task and 1 retry on Commit.DoesNotExist error.

### CODE REVIEW

**Review by leeandher — COMMENTED** [review]
@NisanthanNanthakumar  Code-wise it looks good, just comments and variable name nitpicks from me 👍

File: src/sentry/tasks/codeowners/code_owners_auto_sync.py [hunk 1]
```diff
@@ -1,25 +1,21 @@
...
-from sentry.tasks.base import instrumented_task
+from sentry.tasks.base import instrumented_task, retry
 
 
 @instrumented_task(
     name="sentry.tasks.code_owners_auto_sync",
     queue="code_owners",
-    # If the task fails, the user can manually sync. We'll send them an email on failure.
-    max_retries=0,
+    default_retry_delay=60 * 5,
```
[the remark this answers is not in the document]
Comment (NisanthanNanthakumar) [reply src/sentry/tasks/codeowners/code_owners_auto_sync.py:11]: ehh this is unrelated to the task countdown. I chose 5min arbitrarily.

File: src/sentry/models/commitfilechange.py [hunk 2]
```diff
@@ -54,7 +54,9 @@ def process_resource_change(instance, **kwargs):
 
     # CODEOWNERS file added or modified, trigger auto-sync
     if instance.filename in filepaths and instance.type in ["A", "M"]:
-        code_owners_auto_sync.delay(instance.commit_id)
+        code_owners_auto_sync.apply_async(
+            kwargs={"commit_id": instance.commit_id}, countdown=60 * 5
```
[the remark this answers is not in the document]
Comment (NisanthanNanthakumar) [reply src/sentry/models/commitfilechange.py:58]: I am a little against non-reusable constants. I'll add a comment

[the remark this answers is not in the document]
Comment (leeandher) [reply src/sentry/models/commitfilechange.py:59]: Alright, as long as it's clarified 👍

File: src/sentry/tasks/codeowners/code_owners_auto_sync.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (leeandher) [reply src/sentry/tasks/codeowners/code_owners_auto_sync.py:11]: Ahhh, I see now, this is just the delay after retry, not the initial delay.

**Review by leeandher — APPROVED** [review]
I still support a putting the delay for both as a constant, but it's not worth blocking over 🚢 🚀

**Pull request merged**